mllib/src/main/scala/org/apache/spark/ml/image/ImageSchema.scala

```diff
@@ -1,266 +0,0 @@
-   * @param bytes Image bytes (for example, jpeg)
-   * @return DataFrame Row or None (if the decompression fails)
-   */
-  private[spark] def decode(origin: String, bytes: Array[Byte]): Option[Row] = {
-
-    val img = try {
-      ImageIO.read(new ByteArrayInputStream(bytes))
-    } catch {
-      // Catch runtime exception because `ImageIO` may throw unexcepted `RuntimeException`.
-      // But do not catch the declared `IOException` (regarded as FileSystem failure)
-      case _: RuntimeException => null
-    }
-
-    if (img == null) {
-      None
-    } else {
-      val isGray = img.getColorModel.getColorSpace.getType == ColorSpace.TYPE_GRAY
-      val hasAlpha = img.getColorModel.hasAlpha
-
-      val height = img.getHeight
-      val width = img.getWidth
-      val (nChannels, mode) = if (isGray) {
-        (1, ocvTypes("CV_8UC1"))
-      } else if (hasAlpha) {
-        (4, ocvTypes("CV_8UC4"))
-      } else {
-        (3, ocvTypes("CV_8UC3"))
-      }
-
-      val imageSize = height * width * nChannels
-      assert(imageSize < 1e9, "image is too large")
-      val decoded = Array.ofDim[Byte](imageSize)
-
-      // Grayscale images in Java require special handling to get the correct intensity
-      if (isGray) {
-        var offset = 0
-        val raster = img.getRaster
-        for (h <- 0 until height) {
-          for (w <- 0 until width) {
-            decoded(offset) = raster.getSample(w, h, 0).toByte
-            offset += 1
-          }
-        }
-      } else {
-        var offset = 0
-        for (h <- 0 until height) {
-          for (w <- 0 until width) {
-            val color = new Color(img.getRGB(w, h), hasAlpha)
-            decoded(offset) = color.getBlue.toByte
-            decoded(offset + 1) = color.getGreen.toByte
-            decoded(offset + 2) = color.getRed.toByte
-            if (hasAlpha) {
-              decoded(offset + 3) = color.getAlpha.toByte
-            }
-            offset += nChannels
-          }
-        }
-      }
-
-      // the internal "Row" is needed, because the image is a single DataFrame column
-      Some(Row(Row(origin, height, width, nChannels, mode, decoded)))
-    }
-  }
-
-  /**
-   * Read the directory of images from the local or remote source
-   *
-   * @note If multiple jobs are run in parallel with different sampleRatio or recursive flag,
-   * there may be a race condition where one job overwrites the hadoop configs of another.
-   * @note If sample ratio is less than 1, sampling uses a PathFilter that is efficient but
-   * potentially non-deterministic.
-   *
-   * @param path Path to the image directory
-   * @return DataFrame with a single column "image" of images;
-   *         see ImageSchema for the details
-   */
-  @deprecated("use `spark.read.format(\"image\").load(path)` and this `readImages` will be " +
-    "removed in 3.0.0.", "2.4.0")
-  def readImages(path: String): DataFrame = readImages(path, null, false, -1, false, 1.0, 0)
```

Comment:
only `readImages` is deprecated. Should we remove all?  If https://github.com/apache/spark/pull/25245/files#r306840809 is the case, we should keep other methods.